src/libvirtApi/nodeDevice.js

```diff
@@ -33,47 +33,44 @@ import { call, timeout } from './helpers.js';
+export async function nodeDeviceGet({
     id: objPath,
     connectionName,
 }) {
-    let deviceXmlObject;
-    return call(connectionName, objPath, 'org.libvirt.NodeDevice', 'GetXMLDesc', [0], { timeout, type: 'u' })
-            .then(deviceXml => {
-                deviceXmlObject = parseNodeDeviceDumpxml(deviceXml);
-                deviceXmlObject.connectionName = connectionName;
+    try {
+        const [deviceXml] = await call(connectionName, objPath, 'org.libvirt.NodeDevice', 'GetXMLDesc', [0], { timeout, type: 'u' });
+        const deviceXmlObject = parseNodeDeviceDumpxml(deviceXml);
+        deviceXmlObject.connectionName = connectionName;
 
-                if (deviceXmlObject.path && ["pci", "usb_device"].includes(deviceXmlObject.capability.type)) {
-                    return cockpit.spawn(["udevadm", "info", "--path", deviceXmlObject.path], { err: "message" })
-                            .then(output => {
-                                const nodeDev = parseUdevDB(output);
-                                if (nodeDev && nodeDev.SUBSYSTEM === "pci") {
-                                    deviceXmlObject.pciSlotName = nodeDev.PCI_SLOT_NAME;
-                                    deviceXmlObject.class = nodeDev.ID_PCI_CLASS_FROM_DATABASE;
-                                } else if (nodeDev && nodeDev.SUBSYSTEM === "usb") {
-                                    deviceXmlObject.class = nodeDev.ID_USB_CLASS_FROM_DATABASE;
-                                    deviceXmlObject.busnum = nodeDev.BUSNUM;
-                                    deviceXmlObject.devnum = nodeDev.DEVNUM;
-                                }
+        if (deviceXmlObject.path && ["pci", "usb_device"].includes(deviceXmlObject.capability.type)) {
+            const output = await cockpit.spawn(["udevadm", "info", "--path", deviceXmlObject.path], { err: "message" });
+            const nodeDev = parseUdevDB(output);
+            if (nodeDev && nodeDev.SUBSYSTEM === "pci") {
+                deviceXmlObject.pciSlotName = nodeDev.PCI_SLOT_NAME;
+                deviceXmlObject.class = nodeDev.ID_PCI_CLASS_FROM_DATABASE;
+            } else if (nodeDev && nodeDev.SUBSYSTEM === "usb") {
+                deviceXmlObject.class = nodeDev.ID_USB_CLASS_FROM_DATABASE;
+                deviceXmlObject.busnum = nodeDev.BUSNUM;
+                deviceXmlObject.devnum = nodeDev.DEVNUM;
+            }
+        }
 
-                                return store.dispatch(updateOrAddNodeDevice(deviceXmlObject));
-                            });
-                } else {
-                    return store.dispatch(updateOrAddNodeDevice(deviceXmlObject));
-                }
-            })
-            .catch(ex => console.warn('GET_NODE_DEVICE action failed for path', objPath, ex.toString()));
+        store.dispatch(updateOrAddNodeDevice(deviceXmlObject));
+    } catch (ex) {
+        console.warn('GET_NODE_DEVICE action failed for path', objPath, ex.toString());
```

Comment:
These 2 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1400-20240125-085010-77feb7f2-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)